requests_oauthlib/oauth2_session.py

```diff
@@ -197,7 +197,10 @@ def fetch_token(self, token_url, code=None, authorization_response=None,
                 redirect_uri=self.redirect_uri, username=username,
                 password=password, **kwargs)
 
-        auth = auth or requests.auth.HTTPBasicAuth(username, password)
+        if not auth and username:
```

Comment:
Heh, reading this now I find the operator precedence confusing. Let's do `if (not auth) and username` just to indicate what's going on here.